### ISSUE TITLE
Bug: Fix for `sort-decorators` moves decorators above JSDoc comments

### ISSUE THREAD

**Issue by reduckted** [azat-io/eslint-plugin-perfectionist#399]
### Describe the bug

The fix for the `sort-decorators` rule will move a decorator above the JSDoc comment. 

It seems as though the JSDoc comment is being kept with the directive that immediately follows it, which makes sense for normal comments, but I would expect JSDoc comments to stay before all directives.

ESLint config:
```js
const perfectionist = require('eslint-plugin-perfectionist');
const { config, configs } = require('typescript-eslint');

module.exports = config(configs.recommended, {
    plugins: { perfectionist },
    rules: {
        'perfectionist/sort-decorators': 'warn',
    },
});
```

Original code:
```ts
export class Test {
    /**
     * Test.
     */
    @B()
    @A()
    foo: number;
}
```

Actual fixed code:
```ts
export class Test {
    @A()
    /**
     * Test.
     */
    @B()
    foo: number;
}
```

Expected fixed code:
```ts
export class Test {
    /**
     * Test.
     */
    @A()
    @B()
    foo: number;
}
```

### Code example

```ts
export class Test {
    /**
     * Test.
     */
    @B()
    @A()
    foo: number;
}
```

### ESLint version

v9.15.0

### ESLint Plugin Perfectionist version

v4.1.2

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @reduckted ,
I understand concern even though I think it's a bit context-dependant.

Theoretically, making top-level comments stay where they are should fix the issue. The drawback is that a decorator comment that gets put to the top will now be fixed unless there is already a top-level comment in place.

I'm not sure yet if this would be better to put that behaviour behind an option or not. I suppose that your use-case is probably the most common one among all users.

**Comment by reduckted** [comment]
It's not so much that it's a top-level comment. It's that it's a _JSDoc_ comment. I can't think of any reason why a decorator should appear _above_ a JSDoc comment.

What makes this scenario worse is when you also use the `eslint-plugin-jsdoc` plugin with the `require-jsdoc` rule enabled. There's a [fixer](https://github.com/gajus/eslint-plugin-jsdoc/blob/HEAD/docs/rules/require-jsdoc.md#fixer) for that rule that adds an empty JsDoc comment. What ends up happening is the `sort-decorators` fixer moves the decorator above the JSDoc comment, then the `require-jsdoc` fixer sees that there isn't a JSDoc comment above the decorators, so it adds a blank one. You end up with this mess:

```ts
export class Test {
    /**
     *
     */
    @A()
    /**
     * Test.
     */
    @B()
    foo: number;
}
```

**Comment by azat-io** [comment]
Fixed in 54dfdd9e7efe97f00df8a9a23781d605aaaad61b.
Released in [v4.2.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.2.0).
